### PR TITLE
Add an appCache fallback for browsers without SW support. r=arcturus

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

File: templates/unminified/cache.tpl.html [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<!doctype html>
+<html manifest="manifest.appcache">
+<script defer async src="cache-iframe.js"></script>
+<title>c</title>
```
Comment (gmarty) [inline templates/unminified/cache.tpl.html:4]: This file looks weird but is actually valid HTML5, as per the [HTML5 Validator](http://html5.validator.nu).

Comment (arcturus) [inline templates/unminified/cache.tpl.html:4]: Do we need title?

File: README.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 ========
 
-A [gulp](http://gulpjs.com/) plugin that will automatically generate all the files and infrastructure to make your web app work offline with ServiceWorkers.
+A [gulp](http://gulpjs.com/) plugin to make your web app work offline.
 
 Details
 -------
-This plugin uses the library [ServiceWorkerWare](https://github.com/fxos-components/serviceworkerware) to provide the integration with ServiceWorker.
+This plugin uses the library [ServiceWorkerWare](https://github.com/fxos-components/serviceworkerware) to provide the integration with Service Worker.
+The AppCache manifest is generated by [node-appcache-generator](https://github.com/arcturus/node-appcache-generator).
```
Comment (arcturus) [inline README.md:23]: We can explain that appcache will be used as a fallback if your browser doesn't support service workers

File: README.md [hunk 3]
```diff
@@ -26,6 +26,15 @@ gulp.task('offline', function() {
 });
 ```
 
+How it works
+------------
+Once applied to your app, this plugin code caches all your frontend assets using
+service workers if supported.
+
+If not, AppCache is used. In this mode, an invisible `<iframe>` is appended to
```
Comment (arcturus) [inline README.md]: Oh, ok, is here, forget my previous comment :)

File: templates/unminified/install.tpl.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+(function() {
+  'use strict';
+
+  if (navigator.serviceWorker && !navigator.serviceWorker.controller) {
+    navigator.serviceWorker.register('sw.js');
+  } else {
+    // @see https://github.com/matthew-andrews/workshop-making-it-work-offline/tree/master/05-offline-news/04-more-hacking-appcache
+    const frame = document.createElement('iframe');
```
Comment (arcturus) [inline templates/unminified/install.tpl.js]: This will require >= IE 11 ... but ... YES let's do it!

File: templates/unminified/README.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# JavaScript template files
```
Comment (arcturus) [inline templates/unminified/README.md:1]: This is a great idea, will be easy to read looking here.

File: templates/unminified/cache.tpl.html (hunk 1 above)
Comment (gmarty) [inline templates/unminified/cache.tpl.html:4]: Actually yes, otherwise it's not valid HTML5! (I didn't know about that before).

**Comment by gmarty** [comment]
I've done a lot of changes to take the redirection approach. I tested on a Mac and iPhone.
@arcturus: Can you r? it please?

File: templates/unminified/cache.tpl.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
+(function() {
+  'use strict';
+
+  if (!('URLSearchParams' in window)) {
+    // Simple polyfill for URLSearchParams#has() and URLSearchParams#get().
+    class URLSearchParamsPolyfill {
```
Comment (arcturus) [inline templates/unminified/cache.tpl.js:6]: If we use es6 classes (or other es6 feature not present in webkit browsers) will require a transpilation in the process, forcing people to add that step if they don't have it.

**Comment by arcturus** [comment]
Looks good to me, left a tiny comment.

@gmarty Can you bump the version too in this PR?

File: templates/unminified/cache.tpl.js (hunk 6 above)
Comment (gmarty) [inline templates/unminified/cache.tpl.js:6]: The transpiling is done as part of the minification. If Closure Compiler is used as described in the Readme, then it is also automatically transpiled to es5.

**Pull request merged**